### PR TITLE
Add `writeConnectionSecretToRef` to Go types

### PR DESCRIPTION
**Pull request by glrf**
When interacting with claims through Go we use the types we define in the component.

This PR adds the `writeConnectionSecretToRef` field so we can interact with the connection secret in Go. However this field is mananged by Crossplane, wo strip out the field in Jsonnet before applying it to the XRD.

The PR doesn't change anything about the component output.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [ ] Update the documentation. *Don't forget to generate the CRD API!*
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: apis/vshn/v1/dbaas_vshn_postgresql.go [hunk 1]
```diff
@@ -32,6 +33,9 @@ type VSHNPostgreSQL struct {
 type VSHNPostgreSQLSpec struct {
 	// Parameters are the configurable fields of a VSHNPostgreSQL.
 	Parameters VSHNPostgreSQLParameters `json:"parameters,omitempty"`
+
+	// WriteConnectionSecretToRef references a secret to which the connection details will be written.
+	WriteConnectionSecretToRef corev1.LocalObjectReference `json:"writeConnectionSecretToRef,omitempty"`
```
Comment (glrf) [inline apis/vshn/v1/dbaas_vshn_postgresql.go:38]: This is the actual change I need right now. This allows me to access the connection secret for the SLI exporter

File: crds/appcat.vshn.io_objectbuckets.yaml [hunk 2]
```diff
@@ -60,6 +60,16 @@ spec:
                 - bucketName
                 - region
                 type: object
+              writeConnectionSecretToRef:
+                description: WriteConnectionSecretToRef references a secret to which
+                  the connection details will be written.
+                properties:
+                  name:
+                    description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                      TODO: Add other useful fields. apiVersion, kind, uid?'
```
Comment (zugao) [inline crds/appcat.vshn.io_objectbuckets.yaml:69]: Maybe you missed that

**Review by zugao — APPROVED** [review]
LGTM

File: crds/appcat.vshn.io_objectbuckets.yaml (hunk 2 above)
Comment (glrf) [reply crds/appcat.vshn.io_objectbuckets.yaml:69]: Hmm. Interestingly this comes from k8s's type `corev1.LocalObjectReference` :thinking:

**Pull request merged**